src/main/java/org/jabref/logic/importer/fetcher/ComplexSearchQuery.java

```diff
@@ -0,0 +1,166 @@
+package org.jabref.logic.importer.fetcher;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import org.jabref.model.strings.StringUtil;
+
+public class ComplexSearchQuery {
+    // Field for non-fielded search
+    private final String defaultField;
+    private final List<String> authors;
+    private final List<String> titlePhrases;
+    private final Integer fromYear;
+    private final Integer toYear;
+    private final Integer singleYear;
+    private final String journal;
+
+    private ComplexSearchQuery(String defaultField, List<String> authors, List<String> titlePhrases, Integer fromYear, Integer toYear, Integer singleYear, String journal) {
+        this.defaultField = defaultField;
+        this.authors = authors;
+        this.titlePhrases = titlePhrases;
+        this.fromYear = fromYear;
+        // Some APIs do not support, or not fully support, year based search. In these cases, the non applicable parameters are ignored.
+        this.toYear = toYear;
+        this.journal = journal;
+        this.singleYear = singleYear;
+    }
+
+    public Optional<String> getDefaultField() {
+        return Optional.ofNullable(defaultField);
+    }
+
+    public Optional<List<String>> getAuthors() {
+        return Optional.ofNullable(authors);
+    }
+
+    public Optional<List<String>> getTitlePhrases() {
+        return Optional.ofNullable(titlePhrases);
+    }
+
+    public Optional<Integer> getFromYear() {
+        return Optional.ofNullable(fromYear);
+    }
+
+    public Optional<Integer> getToYear() {
+        return Optional.ofNullable(toYear);
+    }
+
+    public Optional<Integer> getSingleYear() {
+        return Optional.ofNullable(singleYear);
+    }
+
+    public Optional<String> getJournal() {
+        return Optional.ofNullable(journal);
+    }
+
+    public static ComplexSearchQueryBuilder builder() {
+        return new ComplexSearchQueryBuilder();
+    }
+
+    public static class ComplexSearchQueryBuilder {
+        private String defaultField;
+        private List<String> authors;
+        private List<String> titlePhrases;
+        private String journal;
+        private Integer fromYear;
+        private Integer toYear;
+        private Integer singleYear;
+
+        public ComplexSearchQueryBuilder() {
+        }
+
+        public ComplexSearchQueryBuilder defaultField(String defaultField) {
+            if (Objects.requireNonNull(defaultField).isBlank()) {
+                throw new IllegalArgumentException("Parameter must not be blank");
+            }
+            this.defaultField = defaultField;
+            return this;
+        }
+
+        /**
+         * Adds author and wraps it in quotes
+         */
+        public ComplexSearchQueryBuilder author(String author) {
+            if (Objects.requireNonNull(author).isBlank()) {
+                throw new IllegalArgumentException("Parameter must not be blank");
```

Comment:
Is it really a problem if the user searches for an empty string, e.g. `author = ""`?

Reply:
Hey :)
IMHO allowing empty strings for any query parameter does not really make much sense semantically.
I do not see the semantics such an empty string should have:
 - If the user does not want to specify an author, he should just leave the field null
 - Otherwise, he tries to searches for documents without authors?

Did I overlook a use case where giving an empty string for a query parameter is useful?

Reply:
Sounds good! I just wanted to double check. 
(We should keep this in the back of our heads when the user interface is implemented. There you would like to show a helping message instead of an error dialog).